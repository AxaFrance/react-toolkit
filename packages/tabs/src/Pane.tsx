import * as React from 'react';
import { withClassDefault } from '@axa-fr/react-toolkit-core';
import { compose, branch, renderNothing } from 'recompose';

const DEFAULT_CLASSNAME = 'af-tabs__pane';

export type DivProps = React.DetailedHTMLProps<
  React.HTMLAttributes<HTMLDivElement>,
  HTMLDivElement
>;

interface PaneComponentProps extends DivProps {
  active: boolean;
  className?: string;
  children?: React.ReactNode;
}
const Pane: React.SFC<PaneComponentProps> = ({ children, className }) => (
  <div className={className}>{children}</div>
);

const enchance = compose<PaneComponentProps, PaneComponentProps>(
  branch((props: PaneComponentProps) => props.active === false, renderNothing),
  withClassDefault(DEFAULT_CLASSNAME)
)(Pane);

export default enchance;
