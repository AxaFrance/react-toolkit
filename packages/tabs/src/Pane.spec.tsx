import React from 'react';
import { create } from 'react-test-renderer';
import Pane from './Pane';

describe('<Pane>', () => {
  it('should render component', () => {
    const component = create(
      <Pane active>
        <div>Content</div>
      </Pane>
    );
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  });
});
