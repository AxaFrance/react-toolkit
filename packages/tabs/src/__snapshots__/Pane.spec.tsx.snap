// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pane> should render component 1`] = `
<div
  className="af-tabs__pane"
>
  <div>
    Content
  </div>
</div>
`;
